components/merkledb/src/db.rs

```diff
@@ -337,28 +387,49 @@ pub enum Change {
 /// [`rollback`] methods), which allows rolling back some of the latest changes (e.g., after
 /// a runtime error). Checkpoint is created automatically after calling the `flush` method.
 ///
-/// `Fork` implements the [`Snapshot`] trait and provides methods for both reading and
-/// writing data. Thus, `&mut Fork` is used as a storage view for creating
-/// read-write indices representation.
+/// `Fork` provides methods for both reading and writing data. Thus, `&Fork` is used
+/// as a storage view for creating read-write indices representation.
 ///
 /// **Note.** Unless stated otherwise, "key" in the method descriptions below refers
 /// to a full key (a string column family name + key as an array of bytes within the family).
 ///
-/// **Note.** It is possible to create only one instance of index with the specified name based on a
-/// single fork. This restriction is due to impossibility to obtain multiple mutable references to
-/// the same change set inside the fork.
+/// # Borrow checking
+///
+/// It is possible to create only one instance of index with the specified `IndexAddress` based on a
+/// single fork. If an additional instance is requested, the code will panic in runtime.
+/// Hence, obtaining indexes from a `Fork` functions similarly to [`RefCell::borrow_mut()`].
 ///
 /// For example the code below will panic at runtime.
 ///
-/// ```rust, no_run
-/// use exonum_merkledb::{TemporaryDB, ListIndex, Database};
+/// ```rust,should_panic
+/// # use exonum_merkledb::{access::AccessExt, TemporaryDB, ListIndex, Database};
 /// let db = TemporaryDB::new();
 /// let fork = db.fork();
-///
-/// let index1: ListIndex<_, u8> = ListIndex::new("index", &fork);
+/// let index = fork.as_ref().get_list::<_, u8>("index");
 /// // This code will panic at runtime.
-/// let index2: ListIndex<_, u8> = ListIndex::new("index", &fork);
+/// let index2 = fork.as_ref().get_list::<_, u8>("index");
+/// ```
+///
+/// To enable immutable / shared references to indexes, you may use [`readonly`] method:
+///
 /// ```
+/// # use exonum_merkledb::{access::AccessExt, TemporaryDB, ListIndex, Database};
+/// let db = TemporaryDB::new();
+/// let fork = db.fork();
+/// fork.as_ref().get_list::<_, u8>("index").extend(vec![1, 2, 3]);
+///
+/// let readonly = fork.readonly();
```

Comment:
What about naming like `RwLock::read`? 

```rust
let readonly = fork.read();
```

Or 

```rust
let rw = fork.write();
```

To replace `fork.as_ref()`?